static/js/stream_edit.js

```diff
@@ -387,8 +388,10 @@ function show_subscription_settings(sub) {
     sort_but_pin_current_user_on_top(users);
 
     function get_users_for_subscriber_typeahead() {
-        const potential_subscribers = peer_data.potential_subscribers(stream_id);
-        return user_pill.filter_taken_users(potential_subscribers, pill_widget);
+        let potential_subscribers = peer_data.potential_subscribers(stream_id);
+        potential_subscribers = user_pill.filter_taken_users(potential_subscribers, pill_widget);
+        // Exclude muted users from add-subscriber typeahead.
+        return potential_subscribers.filter((user) => !muting.is_user_muted(user.user_id));
```

Comment:
I'm not sure we want to exclude the muted user from this widget, since it's part of the settings interface?

Reply:
Ok, dropped this commit.